PR: https://github.com/odoo/odoo/pull/146405

From: 59f055a87962626dfe5ced26074d43c6989dcf8c
From: vlst
Date: 2023-12-15 09:18:57

Structural Changes: 6
Total Changes: 6

[FIX] pos_self_order: remove blank option

In this pr we remove the blank choice in the self-ordering mode select.
It's unnecessary and throws a validation error on saving settings.

Task 3599144

closes odoo/odoo#146405

X-original-commit: 7403a2b7a7dd8e9d79a33bfeb5c28f6cf1db2ea6
Signed-off-by: Robin Heinz (rhe) <rhe@odoo.com>
Signed-off-by: Vlad Stroia (vlst) <vlst@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_self_order/models/res_config_settings.py
+++ b/addons/pos_self_order/models/res_config_settings.py
@@ -14,15 +14,15 @@ class ResConfigSettings(models.TransientModel):
     _inherit = "res.config.settings"
 
     pos_self_ordering_takeaway = fields.Boolean(related="pos_config_id.self_ordering_takeaway", readonly=False)
-    pos_self_ordering_service_mode = fields.Selection(related="pos_config_id.self_ordering_service_mode", readonly=False)
-    pos_self_ordering_mode = fields.Selection(related="pos_config_id.self_ordering_mode", readonly=False)
+    pos_self_ordering_service_mode = fields.Selection(related="pos_config_id.self_ordering_service_mode", readonly=False, required=True)
+    pos_self_ordering_mode = fields.Selection(related="pos_config_id.self_ordering_mode", readonly=False, required=True)
     pos_self_ordering_alternative_fp_id = fields.Many2one(related="pos_config_id.self_ordering_alternative_fp_id", readonly=False)
     pos_self_ordering_default_language_id = fields.Many2one(related="pos_config_id.self_ordering_default_language_id", readonly=False)
     pos_self_ordering_available_language_ids = fields.Many2many(related="pos_config_id.self_ordering_available_language_ids", readonly=False)
     pos_self_ordering_image_home_ids = fields.Many2many(related="pos_config_id.self_ordering_image_home_ids", readonly=False)
     pos_self_ordering_image_brand = fields.Image(related="pos_config_id.self_ordering_image_brand", readonly=False)
     pos_self_ordering_image_brand_name = fields.Char(related="pos_config_id.self_ordering_image_brand_name", readonly=False)
-    pos_self_ordering_pay_after = fields.Selection(related="pos_config_id.self_ordering_pay_after", readonly=False)
+    pos_self_ordering_pay_after = fields.Selection(related="pos_config_id.self_ordering_pay_after", readonly=False, required=True)
     pos_self_ordering_default_user_id = fields.Many2one(related="pos_config_id.self_ordering_default_user_id", readonly=False)
 
     @api.onchange("pos_self_ordering_default_user_id")
